src/test/StdCheats.t.sol

```diff
@@ -247,6 +247,13 @@ contract StdCheatsTest is Test {
         Receipt[] memory receipts = readReceipts(path);
     }
 
+    function testAssertions() public {
+        assertUint32Eq(uint32(1), uint32(1));
+        assertUint64Eq(uint64(1), uint64(1));
+        assertUint96Eq(uint96(1), uint96(1));
+        assertUint128Eq(uint128(1), uint128(1));
```

Comment:
I was going to comment on changing all the names to `assertEq`, then realized solc errors due to ambiguity. So then I deleted the 4 new assertions and changed these 4 tests to just use `assertEq`, and as expected solc upcasts them to uint256 using the existing assertions. So now I'm struggling to remember why we need these 😅

To clarify, if you remove the changes in `Test.sol`, all of these pass:

```solidity
function testAssertions() public {
    assertEq(uint32(1), uint32(1));
    assertEq(uint64(1), uint64(1));
    assertEq(uint96(1), uint96(1));
    assertEq(uint128(1), uint128(1));
}
```

Reply:
While coding in 0.7.6, I have to explicitly cast the uint32 to uint256 to use assert.

Reply:
Didn't know that they are automatically upcasted in >0.8 . It's been so long in 0.7.0 land

Reply:
Heh sorry to hear that 😛